server/pull_request.go

```diff
@@ -457,6 +468,44 @@ func (s *Server) CleanOutdatedPRs() {
 	mlog.Info("Finished update the outdated prs in the mattermod database....")
 }
 
+func (s *Server) CleanUpLabels(client *GithubClient, pr *model.PullRequest) {
+	labelsToRemove := []string{
+		"AutoMerge",
+		"Awaiting Submitter Action",
+		"Do Not Merge/Awaiting Loadtest",
+		"Do Not Merge/Awaiting Next Release",
+		"Do Not Merge/Awaiting PR",
+		"Do Not Merge",
+		"Setup Cloud Test Server",
+		"Setup HA Cloud Test Server",
+		"Setup Old Test Server",
+		"Setup Upgrade Test Server",
+		"Work In Progress",
+	}
+
+	labels, _, err := client.Issues.ListLabelsByIssue(context.Background(), pr.RepoOwner, pr.RepoName, pr.Number, nil)
+	if err != nil {
+		mlog.Error("Error listing the labels for closed PR", mlog.Err(err))
+		return
+	}
+
+	var wg sync.WaitGroup
```

Comment:
I would recommend using `var g errgroup.Group` here instead of this as that will help with the easier transition to when the go-routine functions start returning errors. It'll just set it up nicely for that. Otherwise someone will have to add the error handling logic here, which `errorgroup.Group` already does inside.

https://www.oreilly.com/content/run-strikingly-fast-parallel-file-searches-in-go-with-sync-errgroup/

Code looks much more clean. Just a suggestion :)

Reply:
seeing our discussion above, no major advantage of using `errorgroup` then since we don't plan to return any errors in the near future.